### PR TITLE
Harrison/pipeline prompt

### PR DESCRIPTION
**Pull request by hwchase17**
idea is to make prompts more composable

### CODE REVIEW

File: langchain/prompts/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                kwargs[k] = prompt.format(**_inputs)
+        _inputs = _get_inputs(kwargs, self.final_prompt.input_variables)
+        return self.final_prompt.format_prompt(**_inputs)
+
+    def format(self, **kwargs: Any) -> str:
+        return self.format_prompt(**kwargs).to_string()
+
+    @property
+    def _prompt_type(self) -> str:
+        raise ValueError
```
Comment (nfcampos) [inline langchain/prompts/pipeline.py:43]: we don't want this?

**Review by nfcampos — APPROVED** [review]
lgtm

File: langchain/prompts/pipeline.py (hunk 1 above)
Comment (hwchase17) [reply langchain/prompts/pipeline.py:43]: probably eventually but ive been defaulting to not including until we figure out serialization

Comment (nfcampos) [reply langchain/prompts/pipeline.py:43]: Okay ive actually started using this for serialisation

**Pull request merged**